### DISCUSSION TITLE
Notification Url receiving multiple notifications and blank body

### DISCUSSION THREAD

**Discussion by BenGWeeks** [btcpayserver/btcpayserver#3795] [General]
Hi there,

I am implementing a simple PoC on a Raspberry Pi that uses the GPIO to control a solenoid for the dispensing of beer. It looks like I could use your POS as a way for users to choose a drink, and accept payment, such as:

![image](https://user-images.githubusercontent.com/5950171/171075607-8be89f8f-3311-424f-aaae-dafa8acc820d.png)

I was then going to use the callback to initiate a Node.js function to dispense the drink.

Pretty sure it was behaving at first (i.e. the callback would fire off a single time after BTCPayServer received a Lightning payment). However, I am getting multiple POSTs sent to the Notification Url. Is this a known issue?

![image](https://user-images.githubusercontent.com/5950171/171075416-82e1f5f6-5f2e-4e69-bff4-c0e9296fce13.png)

Also, the POST body seems to be undefined? Was expecting the JSON package with ID, price, etc. This could well be a "programmer issue" so help is much appreciated!

![image](https://user-images.githubusercontent.com/5950171/171075985-415d6197-747a-44d4-a530-9b6a05d3397d.png)

Even if I did retrieve the POST data, it doesn't look like it will send me what was purchased from the "Point of Sale" app even. I need to know whether to serve a large or small beer?! ;-) ref:

```
{
  "id":"SkdsDghkdP3D3qkj7bLq3",
  "url":"https://btcpay.example.com/invoice?id=SkdsDghkdP3D3qkj7bLq3",
  "status":"paid",
  "price":10,
  "currency":"EUR",
  "invoiceTime":1520373130312,
  "expirationTime":1520374030312,
  "currentTime":1520373179327,
  "exceptionStatus":false,
  "buyerFields":{
    "buyerEmail":"customer@example.com",
    "buyerNotify":false
  },
  "paymentSubtotals": {
    "BTC":114700
  },
  "paymentTotals": {
    "BTC":118400
  },
  "transactionCurrency": "BTC",
  "amountPaid": "1025900",
  "exchangeRates": {
    "BTC": {
      "EUR": 8721.690715789999,
      "USD": 10817.99
    }
  }
}
```

NB Maybe I will get the items purchased from https://btcpay.example.com/invoices/{invoiceId}?

Lastly, is there any way to allow the user to go back to the PoS items without paying (perhaps they realise they chose the wrong size). At the moment they need to use the browser back or wait for it to time out, but I want this page to be full screen.

Reference: https://github.com/BenGWeeks/BTCPayServer-LightningBeerTap

**Comment by BenGWeeks** [comment]
Any ideas on this one @pavlenex?

**Reply by BenGWeeks** [reply]
Looking at your webhook documentation, https://docs.btcpayserver.org/API/Greenfield/v1/, I wonder whether I am getting multiple notifications coming in as didn't respond with res.writeHead(200), or do you need another specific response?

**Reply by bolatovumar** [reply]
@BenGWeeks you can go to the invoice details page and there is a "Webhooks" section which should show all of the webhook delivery attempts. I'm guessing it could be that the status of the invoice changed multiple times hence why the webhook was delivered multiple times.

![Capture](https://user-images.githubusercontent.com/1934678/177830792-b4fdf216-7c15-44a7-87cb-d143c5396214.PNG)

> Lastly, is there any way to allow the user to go back to the PoS items without paying (perhaps they realise they chose the wrong size). At the moment they need to use the browser back or wait for it to time out, but I want this page to be full screen.

No, there is not. Could you raise this as a separate discussion as a feature request? We could make this work by including a back button on the invoice page or the invoice itself and have it as a separate option you can enable on your app if there is enough support for this.
